A series of minor validator test changes

kmaglione opened the pull request:
See linked bugs for details, if necessary.

magopian commented on validator/testcases/javascript/call_definitions.py:
```diff
@@ -167,6 +167,26 @@ def spellcheck_updatecurrentdictionary(wrapper, arguments, traverser):
+    """
+    traverser.err.warning(
+        err_id=("testcases_javascript_call_definititions",
+         "webbrowserpersist"),
```
Strange indendation, pretty sure flake8 will complain

kmaglione replied:
Yeah. I copied that from from the test below and decided to just err on the side of consistency, but I'll just change them both. (flake8 incidentally complains so much about this file that looking at the output is barely even useful)

magopian commented on validator/testcases/javascript/predefinedentities.py:
```diff
@@ -267,7 +300,12 @@
+                        "at shutdown."}},
+         "dangerous": lambda a, t, e:
+            lambda t, e: (e.metadata.get("is_jetpack") and
+                "The observer service should not be used directly in SDK "
```
Is flake8 happy with this indentation?

kmaglione replied:
Nope.

magopian commented on validator/testcases/regex.py:
```diff
@@ -334,6 +362,23 @@ def rewrap():
+
+    def js_tests(self):
+        yield self.get_test(
+                r"@mozilla\.org/extensions/manager;1|"
```
To much indentation

kmaglione replied:
Hm. This is the indentation used for every other test in this file (except for a couple of others that I added in this PR). I think they should all be consistent, so should I leave these as-is, or fix all the others?

magopian replied:
Yeah, I guess leave it like that, at least for now, maybe one day i'll do a "flake8 pass" on all of amo-validator. Thanks @kmaglione

magopian commented:
I don't have the knowledge regarding the rules/warnings/tests added here, who does and could review this?

Also @kmaglione please rebase this branch now that I've merged some of your other PRs, thanks!

kmaglione commented:
I don't think there's currently anyone who's especially qualified to review the code at the moment. @jvillalobos can review the semantics, I suppose.

kmaglione commented:
Rebased and fixed some of the indentation problems. I'd accidentally added the synchronous SQL tests both here and in #256, so I had to merge the functionality of the two.

magopian commented:
Thanks a lot for all this work @kmaglione
